games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+        ?.setInteractive()
+        .on('pointerdown', () => {
+          const card = this.deckFront.pop();
+          card?.playMoveTween(
+            this.playerHandZones[0].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              this.players[0].hand.length -
+              1 * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[0].y - 50
+          );
+          if (card) this.players[0].addCardToHand(card);
+          this.resetHandsMove(this.players[0]);
+          resolve();
+        });
+    });
+  }
+
+  private checkIfCardCanBePutOnMeldZone(): boolean {
+    const playerHand: Card[] = this.players[0].hand;
+    const rankCounts: Record<string, number> = {};
+    playerHand.forEach((card) => {
+      if (rankCounts[card.rank]) rankCounts[card.rank] += 1;
+      else rankCounts[card.rank] = 1;
+    });
+
+    const sameRankCards: Card[] = Object.keys(rankCounts)
+      .map((rank) => {
+        if (rankCounts[rank] >= 3) {
+          return playerHand.find(
+            (card) => card.rank === rank
+          );
+        }
+        return null;
+      })
+      .filter((card) => card !== null) as Card[];
+
+    const suits: string[] = Array.from(
+      new Set(playerHand.map((card) => card.suit))
+    );
+    const consecutiveRankCards: Card[] = [];
+    suits.forEach((suit) => {
+      const sortedHand = playerHand
+        .filter((card) => card.suit === suit)
+        .sort(
+          (a, b) =>
+            a.getRankNumber('rummy') -
+            b.getRankNumber('rummy')
+        );
+      let temp: Card[] = [sortedHand[0]];
+      for (let i = 1; i < sortedHand.length; i += 1) {
+        if (
+          sortedHand[i].getRankNumber('rummy') -
+            sortedHand[i - 1].getRankNumber('rummy') ===
+          1
+        ) {
+          temp.push(sortedHand[i]);
+        } else {
+          consecutiveRankCards.push(
+            ...temp.slice(0, Math.min(temp.length, 3))
+          );
+          temp = [sortedHand[i]];
+        }
+      }
+      consecutiveRankCards.push(
+        ...temp.slice(0, Math.min(temp.length, 3))
+      );
+    });
+
+    const canPutOnMeldZone =
+      sameRankCards.length >= 3 ||
+      consecutiveRankCards.length >= 3;
+
+    return canPutOnMeldZone;
+  }
+
+  private async houseTurn() {
+    // 次にまた、houseTurn()するかどうか
+    if (this.deckBack.length < 1) {
+      this.resetDeckFromDeckFront();
+    }
+    let counter = 0;
+    this.addPutOnHand(this.players[1]);
+    // meldZoneにカードを置けるかチェック
+    const [sameRankCards, consecutiveRankCards] =
+      this.findMeldAbleCard(this.players[1].hand);
+
+    console.log(sameRankCards);
+    console.log(consecutiveRankCards);
+
+    // 追跡用
+    if (sameRankCards)
+      this.meldZoneSameRank.push(...sameRankCards);
+    if (consecutiveRankCards)
+      this.meldZoneConsecutiveRanks.push(
+        ...consecutiveRankCards
+      );
+
+    if (
+      sameRankCards.length > 0 ||
+      consecutiveRankCards.length > 0
+    ) {
```

Comment:
冗長ですので、以下のように変更してください。また、以下の条件は、playerTurn関数の中でも使用されているので、hasMeldableCardsという関数にしたらどうでしょうか？

private hasMeldableCards(): boolean {
    return sameRankCards.length || consecutiveRankCards.length;
}

```suggestion
    if (sameRankCards.length || consecutiveRankCards.length) {
```